src/js/jquery/jquery.js

```diff
@@ -366,17 +347,24 @@ function initiateAction(actionCode, options) {
         ajax_options.data = JSON.stringify(args)
     }
     jQuery.ajax(url, ajax_options);
+
+    return false;
 }
 
-function initiateFormAction(actionCode, form, options) {
-    // Hidden "action" field should not be serialized on AJAX
+function initiateFormAction(actionCode, event, eventThis, options) {
+    const form = $(eventThis);
     var options = options || {};
-    var action_arguments = form.serializeObjectWithSubmit();
+    var action_arguments = form.serializeObject();
+    if (["BUTTON", "INPUT"].includes(event.submitter && event.submitter.tagName)) {
```

Comment:
this could probably be even more generic as just a

`if (event.submitter && "name" in event.submitter && "value" in event.submitter)`

but catching all <input> and <button> is a decent start